### PR TITLE
Move text_rank implementation to package and rerun examples

### PR DESCRIPTION
**Pull request by PrimozGodec**
Textrank does not work correctly in case we want to keep accentuations (including carons) - does not build connections to words with accentuation and those they are not between keywords. It could be fixed in Gensim but they deprecated and removed the module. It will not be available in the next release anymore.

I then searched for other implementations of Text Rank but the issue is that all other implementations support just major languages (excluding Slovenian). Their support for the language is just required for preprocessing but they do not offer the option for manual preprocessing. So there is no other option than to implement (or move implementation) here.  

In this PR:
- I included the Gensim implementation of Text Rank in this library
- Fixed the described error
- Rerun all notebooks that use Text rank. Results are better now.

### CODE REVIEW

File: textsemantics/textrank/textcleaner.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    Returns
+    -------
+    generator
+        Generator that yields sequence words of the given text.
+
+    Example
+    -------
+    .. sourcecode:: pycon
+
+        >>> from gensim.summarization.textcleaner import tokenize_by_word
```
Comment (ajdapretnar) [inline textsemantics/textrank/textcleaner.py:176]: gensim?

File: textsemantics/textrank/textcleaner.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    replacement = r"\1" + separator + r"\2"
+    result = text
+    for regex in regexs:
+        result = regex.sub(replacement, result)
+    return result
+
+
+def merge_syntactic_units(original_units, filtered_units, tags=None):
+    """Process given sentences and its filtered (tokenized) copies into
+    :class:`~gensim.summarization.syntactic_unit.SyntacticUnit`. Also adds tags if they are provided to produced units.
```
Comment (ajdapretnar) [inline textsemantics/textrank/textcleaner.py:70]: SyntacticUnit is probably from textsemantics.textrank.syntactic_unit?

File: textsemantics/textrank/pagerank_weighted.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+.. [1] https://en.wikipedia.org/wiki/PageRank
+
+Examples
+--------
+
+Calculate Pagerank for words
+
+.. sourcecode:: pycon
+
+    >>> from gensim.summarization.keywords import get_graph
```
Comment (ajdapretnar) [inline textsemantics/textrank/pagerank_weighted.py:16]: This would probably also be our version?

File: textsemantics/textrank/keywords.py [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+
+.. sourcecode:: pycon
+
+    >>> from textsemantics.textrank import keywords
+    >>> text = '''Challenges in natural language processing frequently involve
+    ... speech recognition, natural language understanding, natural language
+    ... generation (frequently from formal, machine-readable logical forms),
+    ... connecting language and machine perception, dialog systems, or some
+    ... combination thereof.'''
+    >>> keywords(text).split('\\n')
```
Comment (ajdapretnar) [inline textsemantics/textrank/keywords.py:19]: One \ too many.

File: textsemantics/textrank/keywords.py [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+    [u'natural language', u'machine', u'frequently']
```
Comment (ajdapretnar) [inline textsemantics/textrank/keywords.py:20]: I get normal strings, not Unicode. Is it important?

File: textsemantics/textrank/graph.py [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+class Graph (based on IGraph) which implements undirected graph.
+
+Examples
+--------
+
+Create simple graph with 4 nodes.
+
+.. sourcecode:: pycon
+
+    >>> g = Graph()
```
Comment (ajdapretnar) [inline textsemantics/textrank/graph.py:14]: `from textsemantics.textrank.graph import Graph`

File: textsemantics/textrank/commons.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            graph.add_node(item)
+    return graph
+
+
+def remove_unreachable_nodes(graph):
+    """Removes unreachable nodes (nodes with no edges), inplace.
+
+    Parameters
+    ----------
+    graph : :class:`~gensim.summarization.graph.Graph`
```
Comment (ajdapretnar) [inline textsemantics/textrank/commons.py:60]: textsemantics.textrank.graph.Graph?

File: textsemantics/textrank/commons.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+
+Examples
+--------
+
+Create simple graph and add edges. Let's take a look at nodes.
+
+.. sourcecode:: pycon
+
+    >>> gg = build_graph(['Felidae', 'Lion', 'Tiger', 'Wolf'])
```
Comment (ajdapretnar) [inline textsemantics/textrank/commons.py:15]: `from textsemantics.textrank.commons import build_graph`

File: textsemantics/textrank/keywords.py (hunk 5 above)
Comment (PrimozGodec) [reply textsemantics/textrank/keywords.py]: It is ok. In python3 strings are Unicode without the `u'` prefix. I think it is there because of Python2.

**Pull request merged**